sklearn/preprocessing/data.py

```diff
@@ -2348,30 +2360,39 @@ def _transform_col(self, X_col, quantiles, inverse):
             #  for inverse transform, match a uniform PDF
             X_col = output_distribution.cdf(X_col)
         # find index for lower and higher bounds
-        lower_bounds_idx = (X_col - BOUNDS_THRESHOLD <
-                            lower_bound_x)
-        upper_bounds_idx = (X_col + BOUNDS_THRESHOLD >
-                            upper_bound_x)
-
+        # comparison with NaN will raise a warning which we make silent
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            lower_bounds_idx = (X_col - BOUNDS_THRESHOLD <
+                                lower_bound_x)
+            upper_bounds_idx = (X_col + BOUNDS_THRESHOLD >
+                                upper_bound_x)
+
+        mask_finite = np.isnan(X_col)
```

Comment:
I would rather call that `nan_mask`. `mask_finite` sounds like a verb instead of a noun. Furthermore `True` means `nan`, not `finite`.

Reply:
Actually better do:

```python
isfinite_mask = ~np.isnan(X_col)
X_col_finite = X_col[isfinite_mask]
```